jaxrs-spec/src/main/asciidoc/chapters/resources/_declaring_method_capabilities.adoc

```diff
@@ -126,3 +129,58 @@ the media type `application/xml` given its higher qs-value. Note that
 qs-values, just like q-values, are relative and as such are only
 comparable to other qs-values within the same `@Produces` annotation
 instance. For more information see <<determine_response_type>>.
+
+[[consuming_multipart_formdata]]
+==== Special handling for consuming multipart/form-data media types
+
+The `multipart/form-data` media type enables requests to send multiple
+entities (parts) as a single entity. Each part contains it's own set of
+headers, media type and content. There are a few different ways to
+consume these requests in a resource method. First, a resource method
+may contain an entity parameter of type, `List<Part>`. For example:
+
+[source,java]
+----
+@Path("widgets3")
+public class WidgetsResource3 {
+
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public Response postWidget(List<Part> parts) {
+        for (Part part : parts) {
+            String name = part.getName();
+            Optional<String> fileName = part.getFileName();
+            InputStream is = part.getEntityStream();
+            MultivaluedMap<String, String> partHeaders = part.getHeaders();
+            MediaType mediaType = part.getMediaType();
+            doSomethingWithPart(name, fileName, is, partHeaders, mediaType);
+        }
+        return Response.ok().build();
+    }
+
+}
+----
+
+Another approach is to use `@FormParam` parameters where the value in the
+annotation corresponds to the name of the part. The parameter type may be
+a `jakarta.ws.rs.ext.Part`, a `java.io.InputStream` or a `String`. Here is
+an example:
+
+[source,java]
+----
+@Path("widgets4")
+public class WidgetsResource4 {
+
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public Response postWidget(@FormParam("part1Name") String part1, 
+                               @FormParam("part2Name") InputStream part2,
+                               @FormParam("part3Name") Part part3) {...}
+}
+----
+
+The only way to access the headers for a particular part is to use the `Part`
```

Comment:
Why don't we allow to invoke parameter handlers?

Reply:
Sorry, I don't understand your question.